src/mixins/canvas_events.mixin.js

```diff
@@ -146,11 +146,10 @@
       this._hoveredTarget = null;
       target && target.fire('mouseout', { e: e });
 
-      var _this = this;
       this._hoveredTargets.forEach(function(_target){
-        _this.fire('mouse:out', { target: target, e: e });
-        _target && target.fire('mouseout', { e: e });
-      });
+        this.fire('mouse:out', { target: _target, e: e });
+        _target && _target.fire('mouseout', { e: e });
+      }, this);
```

Comment:
since this was probably a distraction/typo as you say, we should name those things better.
I would call 
https://github.com/fabricjs/fabric.js/pull/8011/files#diff-6011216a37d34f94eb2a73dbf67b8c7f58c35b42ba7e882da0aaac6e27225f28R144
as hoveredTarget, and _target hoveredNestedTarget.

What do you think?